### PR TITLE
calendar redesign

### PR DESCRIPTION
**Pull request by demkalkov**

### CODE REVIEW

File: mobile/src/calendar/calendar-legend.tsx [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+    private getVisibleCalendarEvents() {
+        const { intervals, weeks } = this.props;
+        const calendarEventsMap: Map<CalendarEventsType, number> = new Map();
+
+        if (!weeks || !intervals) {
+            return calendarEventsMap;
+        }
+
+        weeks.forEach(week => {
```
Comment (serjKim) [inline mobile/src/calendar/calendar-legend.tsx]: I would put this logic to ```IntervalsModel```, for example to ```eventTypesByMonth()```

File: mobile/src/tabbar/tab-navigator.tsx [hunk 2]
```diff
@@ -13,6 +13,7 @@ const tabNavigatorConfig: TabNavigatorConfig = {
         showIcon: true,
     },
     animationEnabled: false,
+    initialRouteName: 'Calendar'
```
Comment (serjKim) [inline mobile/src/tabbar/tab-navigator.tsx]: Revert )

File: mobile/src/calendar/styles.ts [hunk 3]
```diff
@@ -119,74 +140,65 @@ export const agendaStyles = StyleSheet.create({
...
-export const daysCounterStyles = StyleSheet.create({
+export const selectedDayStyles = StyleSheet.create({
     container: {
-        backgroundColor: daysCounterPrimaryColor,
-        flexDirection: 'column',
-        flex: 1
-    },
-    content: {
-        flexDirection: 'column',
+        //transform: [{ translateX: -(circleDiameter / 2) }],
```
Comment (serjKim) [inline mobile/src/calendar/styles.ts]: Remove commented code

**Comment by serjKim** [comment]
Merge to master

File: mobile/src/calendar/styles.ts [hunk 4]
```diff
@@ -83,18 +83,39 @@ export const calendarStyles = StyleSheet.create({
...
+
+            default:
+                return null;
+        }
+    }
+}
 
 // serjKim: Depends on count of overlapped intervals?..
-const intervalOpacity = 0.5;
+const intervalOpacity = .95;
```
Comment (serjKim) [inline mobile/src/calendar/styles.ts]: ?

Comment (demkalkov) [reply mobile/src/calendar/styles.ts]: When opacity is .5 the colors are too pale and do not match the design at all

File: mobile/src/calendar/calendar-legend.tsx (hunk 1 above)
Comment (demkalkov) [reply mobile/src/calendar/calendar-legend.tsx]: Isn't it view specific logic that might not be required anywhere else?

File: mobile/src/calendar/styles.ts (hunk 4 above)
Comment (serjKim) [reply mobile/src/calendar/styles.ts]: But it's difficult to determine when some overlapped interval is started or finished

File: mobile/src/calendar/calendar-legend.tsx (hunk 1 above)
Comment (serjKim) [reply mobile/src/calendar/calendar-legend.tsx]: I think it looks like a part of IntervalsModel api. And doesn't depend on whether you are going to reuse it or not, having it in component is difficult to test it.

Comment (demkalkov) [reply mobile/src/calendar/calendar-legend.tsx]: @santee

File: mobile/src/calendar/calendar-legend.tsx [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+                <View>
+                    {legend}
+                </View>
+            </View>
+        );
+    }
+
+    private getVisibleCalendarEvents() {
+        const { intervals, weeks } = this.props;
+        const calendarEventsMap: Map<CalendarEventsType, number> = new Map();
```
Comment (santee) [inline mobile/src/calendar/calendar-legend.tsx]: can you put generic parameters in constructor call?

File: mobile/src/calendar/today.tsx [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+import { View } from 'react-native';
+import { connect } from 'react-redux';
+import { DayModel } from '../reducers/calendar/calendar.model';
+import { SelectedDay } from './days-counters/selected-day';
+import { AppState } from '../reducers/app.reducer';
+
+interface TodayProps {
+    selectedCalendarDay: DayModel;
+}
+class TodayImpl extends Component<TodayProps> {
```
Comment (santee) [inline mobile/src/calendar/today.tsx]: Should be removed

**Pull request merged**